pyqtorch/noise.py

```diff
@@ -1,54 +1,47 @@
 from __future__ import annotations
 
-from math import sqrt
+import sys
+from math import log2, sqrt
 
 import torch
 from torch import Tensor
 
-from pyqtorch.apply import operator_product
+from pyqtorch.apply import apply_density_mat
 from pyqtorch.embed import Embedding
-from pyqtorch.matrices import DEFAULT_MATRIX_DTYPE, IMAT, XMAT, YMAT, ZMAT, _dagger
-from pyqtorch.utils import DensityMatrix, density_mat
+from pyqtorch.matrices import DEFAULT_MATRIX_DTYPE, IMAT, XMAT, YMAT, ZMAT
+from pyqtorch.utils import DensityMatrix, density_mat, qubit_support_as_tuple
 
 
 class Noise(torch.nn.Module):
```

Comment:
I think one should save the dtype and overload the to method here to allow dtype conversions, no?